### DISCUSSION TITLE
ai_thinker A202 board & AI_THINKER_ES8388_VOLUME_HACK

### DISCUSSION THREAD

**Discussion by refob** [pschatzmann/arduino-audio-driver#25] [General]
With the arduino-audiokit I had to use "#define AI_THINKER_ES8388_VOLUME_HACK 2" to get decent results on the A202.

With the arduino-audio-driver the "#define AI_THINKER_ES8388_VOLUME_HACK 2" doesn't seem to work.

But using "#define AI_THINKER_ES8388_VOLUME_HACK 1" I now get excellent output using the ai_thinker A202 board.

**Comment by pschatzmann** [comment]
Strange: the code is exactly the same, so I don't expect any difference!

**Comment by kodexolabs** [comment]
It sounds like you've had success optimizing audio output on your AI Thinker A202 board. While `#define AI_THINKER_ES8388_VOLUME_HACK 2` worked well with the arduino-audiokit, it didn't with the arduino-audio-driver. Switching to `#define AI_THINKER_ES8388_VOLUME_HACK 1` yielded excellent results. This could be due to differences in how each library handles the ES8388 codec. Checking the documentation, comparing the source codes, and seeking community insights might further clarify these behaviors and help optimize your setup even more.
